code/game/gamemodes/clock_cult/clock_structures.dm

```diff
@@ -390,15 +390,18 @@
 		return 0
 	flick("[initial(icon_state)]_discharged", src)
 	icon_state = "[initial(icon_state)]_recharging"
-	spawn(3000) //5 minutes
-		if(!src)
-			return 0
-		visible_message("<span class='warning'>The writhing tendrils return to the gemstone, which begins to glow with power.</span>")
-		flick("[initial(icon_state)]_recharged", src)
-		icon_state = initial(icon_state)
-		recharging = FALSE
+	addtimer(src, "recharge", 3000) // 5 minutes
 	return 1
 
+/obj/structure/clockwork/interdiction_lens/proc/recharge()
+	if(!src)
+		return 0
+	visible_message("<span class='warning'>The writhing tendrils return \
+		to the gemstone, which begins to glow with power.</span>")
+	flick("[initial(icon_state)]_recharged", src)
+	icon_state = initial(icon_state)
+	recharging = FALSE
```

Comment:
Can you not touch the lens-related parts because if you do you'll conflict with something I'm putting up today